### DISCUSSION TITLE
fetch command error on parsing parameters - 3.1.30+

### DISCUSSION THREAD

**Discussion by pomareta** [gitpython-developers/GitPython#1568] [Q&A] [answered]
Doing 
![image](https://user-images.githubusercontent.com/6089141/228781137-6d432e8f-3eec-4a1e-a0ec-69c5bdc71b71.png)
repo.remotes.origin.fetch(['--prune', '--prune-tags'])

works well until 3.1.29 version, but when updating to 3.1.30 or 3.1.31 fetch command stops working because it tries to take the parameter --prune as a remote

cmdline: git fetch -v -- origin --prune --prune-tags
stderr: 'fatal: couldn't find remote ref --prune'

![image](https://user-images.githubusercontent.com/6089141/228780141-e2bbc592-e234-4d95-ac01-f9fae25a87ca.png)

Thanks for reviewing this

**Comment by Byron** [comment, marked as answer]
This is a feature to prevent command injection when passing options. I think there are workarounds too, but don't know what exactly it is.

You could try `.fetch(prune=True, prune_tags=True)` which should be except from injection projection, or try to set `.fetch(…, allow_unsafe_options=True`)`.

Hope that helps.

**Comment by pomareta** [comment]
Hello,

Sorry for the late answer. That worked fine. Thanks a lot !
![image](https://github.com/gitpython-developers/GitPython/assets/6089141/f02f39a0-c3c9-4032-b083-96408b0f7513)
